### PR TITLE
Tracing: Add credentialsSecret for basic authentication to remote endpoint

### PR DESCRIPTION
**Pull request by kmjayadeep**
# Changes

In the current implementation, we only supported tracing collectors without authentication. In this PR, an additional field `credentialsSecret` is added to the tracing configmap. The username and password provided in this secret will be used to authenticate against collector endpoint. This allows users to connect to third party opentelemetry compatible collectors.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Tracing config now includes an additional optional field `credentialsSecret` where users can specify the name of a secret. The username and password fields from the secret will be used to authenticate against Tracing collector endpoint.
```

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: pkg/tracing/tracing.go [hunk 1]
```diff
@@ -28,69 +28,119 @@ import (
...
-			logger.Info("Initialized Tracer Provider")
-			if p, ok := t.provider.(*tracesdk.TracerProvider); ok {
-				if err := p.Shutdown(context.Background()); err != nil {
-					logger.Errorf("Unable to shutdown tracingprovider with error : %v", err.Error())
-				}
-			}
-			t.provider = tp
+			creds := sec.Data
+			t.username = string(creds["username"])
+			t.password = string(creds["password"])
```
Comment (Yongxuanzhang) [inline pkg/tracing/tracing.go:85]: can this code be tested?

File: config/controller.yaml [hunk 2]
```diff
@@ -110,13 +110,6 @@ spec:
           value: /etc/ssl/certs
         - name: METRICS_DOMAIN
           value: tekton.dev/pipeline
-       # The following variables can be uncommented with correct values to enable Jaeger tracing
-       #- name: OTEL_EXPORTER_JAEGER_ENDPOINT
-       #  value: http://jaeger-collector.jaeger:14268/api/traces
-       #- name: OTEL_EXPORTER_JAEGER_USER
-       #  value: username
-       #- name: OTEL_EXPORTER_JAEGER_PASSWORD
-       #  value: password
```
Comment (Yongxuanzhang) [inline config/controller.yaml:119]: Does this have conflicts with your current approach? What will happen to users who use these?

File: pkg/reconciler/pipelinerun/controller.go [hunk 3]
```diff
@@ -59,9 +61,10 @@ func NewController(opts *pipeline.Options, clock clock.PassiveClock) func(contex
 		pipelineRunInformer := pipelineruninformer.Get(ctx)
 		resolutionInformer := resolutioninformer.Get(ctx)
 		verificationpolicyInformer := verificationpolicyinformer.Get(ctx)
-		tracerProvider := tracing.New(TracerProviderName)
+		secretinformer := secretinformer.Get(ctx)
```
Comment (Yongxuanzhang) [inline pkg/reconciler/pipelinerun/controller.go:64]: Is it possible to add an e2e test to make sure the secret informorer/lister we added are working properly?

File: config/controller.yaml (hunk 2 above)
Comment (kmjayadeep) [reply config/controller.yaml:119]: These environment variables can be still used. However, if the user provides both the environment variable and configmap option, the  env variables will be ignored.

File: pkg/tracing/tracing.go (hunk 1 above)
Comment (kmjayadeep) [reply pkg/tracing/tracing.go:85]: added a new test

File: pkg/reconciler/pipelinerun/controller.go (hunk 3 above)
Comment (afrittoli) [reply pkg/reconciler/pipelinerun/controller.go:64]: @Yongxuanzhang there is a unit test that exercises this through the secrets fake informer, I think that provides good coverage for that - wdyt?

Comment (afrittoli) [reply pkg/reconciler/pipelinerun/controller.go:64]: It would be good to have a unit test that verifies the case of secret being updated though.

Comment (kmjayadeep) [reply pkg/reconciler/pipelinerun/controller.go:63]: I have added a new test case to cover that - `TestHandlerSecretUpdate`

**Pull request merged**